arcgis-ios-sdk-samples/Edit data/Create and save KML file/CreateAndSaveKMLViewController.swift

```diff
@@ -0,0 +1,270 @@
+    func addPolygon() {
+        let alertController = UIAlertController(title: "Select Color", message: "This color will be used to fill the polygon", preferredStyle: .actionSheet)
+        colors.forEach { (colorTitle, colorValue) in
+                let colorAction = UIAlertAction(title: colorTitle, style: .default) { (_) in
+                self.kmlStyle = self.makeKMLStyleWithPolygonStyle(color: colorValue)
+                self.startSketch(creationMode: .polygon)
+                }
+            alertController.addAction(colorAction)
+        }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        
+        alertController.popoverPresentationController?.barButtonItem = addButton
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    // Make KML with a point style.
+    func makeKMLStyleWithPointStyle(iconURL: URL) -> AGSKMLStyle {
+        let icon = AGSKMLIcon(url: iconURL)
+        let iconStyle = AGSKMLIconStyle(icon: icon, scale: 1.0)
+        kmlStyle.iconStyle = iconStyle
+        return kmlStyle
+    }
+    
+    // Make KML with a line style.
+    func makeKMLStyleWithLineStyle(color: UIColor) -> AGSKMLStyle {
+        kmlStyle.lineStyle = AGSKMLLineStyle(color: color, width: 2.0)
+        return kmlStyle
+    }
+    
+    // Make KML with a polygon style.
+    func makeKMLStyleWithPolygonStyle(color: UIColor) -> AGSKMLStyle {
+        kmlStyle.polygonStyle = AGSKMLPolygonStyle(color: color)
+        kmlStyle.polygonStyle?.isFilled = true
+        kmlStyle.polygonStyle?.isOutlined = false
+        return kmlStyle
+    }
+    
+    // Update the bottom toolbar button.
+    func updateToolbarItems() {
+        guard let sketchEditor = mapView.sketchEditor else {
+            return
+        }
+        let middleButtonItem: UIBarButtonItem
+        if sketchEditor.isStarted {
+            middleButtonItem = sketchDoneButton
+        } else {
+            middleButtonItem = addButton
+        }
+        let flexibleSpace1 = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        let flexibleSpace2 = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        toolbar.items = [resetButtonItem, flexibleSpace1, middleButtonItem, flexibleSpace2, actionButtonItem]
+    }
+    
+    // Start a new sketch mode.
+    func startSketch(creationMode: AGSSketchCreationMode) {
+        mapView.sketchEditor?.start(with: creationMode)
+        updateToolbarItems()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        updateToolbarItems()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "CreateAndSaveKMLViewController",
+            "CreateAndSaveKMLSettingsViewController"
```

Comment:
`CreateAndSaveKMLSettingsViewController` no longer exists, correct?

```suggestion
            "CreateAndSaveKMLViewController"
```